### PR TITLE
Add ability to configure JsonDriver and ObjectDriver

### PR DESCRIPTION
**Pull request by alshenetsky**
This is very handy feature when comparing diffs between snapshots with unicode symbols

### CODE REVIEW

File: src/Drivers/JsonDriver.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ public function extension(): string
     public function match($expected, $actual)
     {
         if (is_array($actual)) {
-            $actual = json_encode($actual, JSON_PRETTY_PRINT)."\n";
+            $actual = json_encode($actual, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE)."\n";
```
Comment (mfn) [inline src/Drivers/JsonDriver.php:32]: There are also other useful options like  JSON_UNESCAPED_SLASHES

Is there a way we can change the driver to:
- have the options configurable
- default to the the pretty print

so it stays backwards compatible?

Strictly speaking, yours is a breaking change for anyone using this package+driver+also having unicode in their output (deliberately or not)

Comment (alshenetsky) [reply src/Drivers/JsonDriver.php:32]: I thought about that too, but then I thought about what we mean by "breaking change." If you look at "phpstan" and "phpcsfixer" packages, they are constantly adding new rules that cause tests to fail after a "composer update". All the user has to do after "composer update" is to update the snapshots/baseline file/ e.t.c.

What do you think? Of course I can totally do it configurable. If so, may be introduce constructor parameters?  Like
```
$this->assertMatchesSnapshot($actual, new JsonDriver(JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT));
```
or something

Comment (mfn) [reply src/Drivers/JsonDriver.php:32]: The thing is, this is an opinionated change: there's no "wrong or right" here.

Some may prefer it with escaping, others may not.

So therefore I consider the forced change the wrong direction and if there's a change, the ability to configure and _keeping_ the current config as default, should be done.

Comment (alshenetsky) [reply src/Drivers/JsonDriver.php]: So I came up with this solution, what do you say?

Comment (alshenetsky) [reply src/Drivers/JsonDriver.php]: @freekmurze sorry for pinging you, but could you approve please?

**Comment by baijunyao** [comment]
Aha, looks like we're doing some duplication. 😂 #131

**Pull request closed without merging**